codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -158,47 +151,49 @@ update these file paths accordingly.
 
   /// The entrance point of examining and code signing an engine artifact.
   Future<void> validateAll() async {
-    for (String passwordFilePath in [
-      codesignAppstoreIDFilePath,
-      codesignTeamIDFilePath,
-      appSpecificPasswordFilePath,
-    ]) {
-      await readPassword(passwordFilePath);
-    }
-    if (availablePasswords.containsValue('')) {
-      throw CodesignException('certian passwords are missing. \n'
-          'make sure you have provided <CODESIGN_APPSTORE_ID>, <CODESIGN_TEAM_ID>, and <APP_SPECIFIC_PASSWORD>');
-    }
-    codesignAppstoreId = availablePasswords['CODESIGN_APPSTORE_ID']!;
-    codesignTeamId = availablePasswords['CODESIGN_TEAM_ID']!;
-    appSpecificPassword = availablePasswords['APP_SPECIFIC_PASSWORD']!;
-
-    await processRemoteZip();
-
-    if (dryrun) {
-      log.info('code signing dry run has completed, If you intend to upload the artifacts back to'
-          ' google cloud storage, please use the --dryrun=false flag to run code signing script.');
+    // The value of codesignedFilePath is used in the cleanup process in the finally block.
+    String? codesignedFilePath;
+    try {
+      for (String passwordFilePath in [
+        codesignAppstoreIDFilePath,
+        codesignTeamIDFilePath,
+        appSpecificPasswordFilePath,
+      ]) {
+        await readPassword(passwordFilePath);
+      }
+      if (availablePasswords.containsValue('')) {
+        throw CodesignException('certian passwords are missing. \n'
+            'make sure you have provided <CODESIGN_APPSTORE_ID>, <CODESIGN_TEAM_ID>, and <APP_SPECIFIC_PASSWORD>');
+      }
+      codesignAppstoreId = availablePasswords['CODESIGN_APPSTORE_ID']!;
+      codesignTeamId = availablePasswords['CODESIGN_TEAM_ID']!;
+      appSpecificPassword = availablePasswords['APP_SPECIFIC_PASSWORD']!;
+      codesignedFilePath = await processRemoteZip();
```

Comment:
if this line is the one generating the temp files you can probably move the try - finally  block here.

Reply:
Sure, moved.